src/main/java/book/store/service/ShoppingCartService.java

```diff
@@ -0,0 +1,15 @@
+package book.store.service;
+
+import book.store.dto.cartitem.CartItemRequestDto;
+import book.store.dto.shoppingcart.ShoppingCartDto;
+import jakarta.validation.constraints.Positive;
+
+public interface ShoppingCartService {
+    ShoppingCartDto addBookInShoppingCart(Long userId, CartItemRequestDto requestDto);
+
+    void updateQuantity(String cartId, @Positive Long cartItemId, int quantity);
```

Comment:
```suggestion
    ShoppingCartDto updateQuantity(String cartId,  Long cartItemId, UpdateCartItemDto .....);
```
It's better to validate in controller, dto and entity